dev/test_mlflow_lazily_imports_pyspark.py

```diff
@@ -0,0 +1,5 @@
+import sys
+import mlflow  # pylint: disable=unused-import
+
+cached_packages = set(k.split(".")[0] for k in sys.modules.keys())
+assert "pyspark" not in cached_packages
```

Comment:
We should do the samething for other frameworks, but the `python-small` check implicitly tests this by importing mlflow in an environment where no ML packages are imported.

Reply:
```
python -c '
import sys
import mlflow
cached_packages = set(k.split(".")[0] for k in sys.modules.keys())
assert "pyspark" not in cached_packages
'
```